apps/src/applab/Visualization.jsx

```diff
@@ -0,0 +1,32 @@
+var applabConstants = require('./constants');
+
+var styles = {
+  hidden: {
+    display: 'none'
+  }
+};
+
+var Visualization = React.createClass({
+  render: function () {
+    var appWidth = applabConstants.APP_WIDTH;
+    var appHeight = applabConstants.APP_HEIGHT - applabConstants.FOOTER_HEIGHT;
+    return (
+      <div>
+        <div id="visualization">
```

Comment:
And protecting here.

These are tricky bugs to catch right now - we have little redux state that is likely to trigger a re-render of this part of the DOM, so you don't end up in a handlers-disconnected state.  Maybe we should write/find a little helper that forces a re-render of part of the DOM to check for this stuff.